### ISSUE TITLE
Stylus button settings doesn't make any difference!

### ISSUE THREAD

**Issue by hamidbhm** [xournalpp/xournalpp#3475]
Hi, I am using Huion Inspiroy H640p with the stable version of Xournal++ .
The settings for stylus buttons doesn't work. I even removed the default settings for keys from graphic tablet driver settings but, the settings of Xournal++ still didn't work.

**Comment by GamerBene19** [comment]
I don't know if my issue is related to this one, but as far as I can tell it is.
I've been using an older Xournal++ version (I don't know which one it was; but I think its irrelevant anyway) on my Laptop (that is running Windows 10) for quite a while now.
A few days ago I did update to the 1.1 hotfix version and with that version the buttons my stylus stopped to work. Pen itself and pressure-sensitivity do still work, only the buttons do not work anymore.
I've downgraded to 1.0.20 for now where the issue is not present.

**Comment by rolandlo** [comment]
@GamerBene19 Are you using a Wacom tablet on Windows 10? Then you may be better off with the non-hotfix version of Xournal++ 1.1.0.

**Comment by lb90** [comment]
Hello everyone 🙂 

I'm currently working on enabling driver button mappings for Windows Ink! It should be ready quite soon.

I'm also working on a preference option for Xournal++ to switch between Wintab and Windows Ink API.

For the time being just continue to use the 1.20 version. I hope to have everything in place for the next Xournal++ release!

Thank you

**Comment by pablolob** [comment]
Hi, 
I am using a one by wacom on windows 10.
 The 1.1.0 and the hotfix recognize the stylus buttons when I press them because the cursor changes correctly but it doesn't draw anything.
Shoud I open an issue or is it the same problem?
Thanks

**Comment by oasido** [comment]
Hey.
I think I'm experiencing the same issue.
Version 1.1.1 (stable), running Windows 11.

I have the Huion 420X and can't get the buttons to work.
Windows Ink option is enabled in the Huion software.

@lb90 Do you have any lead that I can use to solve this?

**Comment by statix531441** [comment]
Hi, 
I'm experiencing the same issue on v1.1.1 in Windows 11 while using One by Wacom.
On button press it changes the tooltip but I can't use it.

**Comment by charstnut** [comment]
Hi,
This issue seems to be still relevant on Win 11, Xournalpp v.1.1.1, with Wacom Intuos S. The symptom is that clicking the pen button would switch to the correct tool but the tool would not work (in my case it's the lasso selection tool). Is there anything I could help with (such as hardware logs, etc.?) Thank you!

**Comment by bhennion** [comment]
See also this comment for a possible fix in driver configuration: https://github.com/xournalpp/xournalpp/issues/3608#issuecomment-1236041732

**Comment by Universal-Invariant** [comment]
I have a Huion GS1331 in Win11 and have issues. 

[I have to disable windows ink in the Huion app or nothing happens.. The stylus essentially does not work work in the app. When it is enabled no buttons work but I can write. If I set right to eraser it works. I think the tablet uses left as "down". This makes it hard to use though. No pressure without ink as it seems to be treating it as a mouse.]


I turned back on ink and restarted xournal++ and it works now but it does not detect stylus buttons. Every once in a while when I push the buttons a lot the square will pop up but it immediately disappears. Sometimes I'll get a random line deleted. If I draw a connected stroke in one go while pressing the buttons the lines because disconnected. The events are clearly getting through but seem to be misread.

In the device list I have DJP Inc. Virtual Multitouch Device for eraser and pen and then I have the HUION Tablet_GS1331 for eraser, touchscreen, and pen.   I've tried disabling various things but nothing seems to work.

Basically the problem now all seems that the buttons are not working.
